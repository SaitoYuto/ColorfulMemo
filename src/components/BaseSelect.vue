<template>
  <v-select
    :bg-color="bgColor"
    :items="items"
    :item-title="itemTitle"
    :item-value="itemTitle"
    :label="label"
    :return-object="returnObject"
    v-model="select"
  ></v-select>
</template>

<script lang="ts" setup>
defineProps({
  bgColor: {
    type: String,
    required: false,
    default: undefined,
  },
  items: {
    type: Array,
    required: true,
  },
  itemTitle: {
    type: String,
    required: false,
    default: "text",
  },
  itemValue: {
    type: String,
    required: false,
    default: "code",
  },
  label: {
    type: String,
    required: true,
  },
  returnObject: {
    type: Boolean,
    required: false,
    default: true,
  },
});

const select = defineModel("select", {
  type: Object,
});
</script>
