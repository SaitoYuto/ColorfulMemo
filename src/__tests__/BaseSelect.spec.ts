import BaseSelect from "@/components/BaseSelect.vue";
import { mount } from "@vue/test-utils";
import "@/__tests__/setup";

describe("BaseSelect.vue", () => {
  it("renders props.label when passed", async () => {
    const label = "Select Label";
    const items = [1, 2, 3, 4];
    const wrapper = mount(BaseSelect, {
      props: {
        label: label,
        items: items,
      },
    });
    expect(wrapper.props("label")).toBe(label);
    expect(wrapper.props("items")).toStrictEqual(items);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders props.text when an empty string passed", async () => {
    const emptyLabel = "";
    const items = [1, 2, 3, 4];
    const wrapper = mount(BaseSelect, {
      props: {
        label: emptyLabel,
        items: items,
      },
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.props("label")).toBe(emptyLabel);
    expect(wrapper.props("items")).toStrictEqual(items);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
