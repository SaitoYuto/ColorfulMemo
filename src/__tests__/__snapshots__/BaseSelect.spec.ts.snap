// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseSelect.vue > renders props.label when passed 1`] = `
"<div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field v-select v-select--single">
  <!---->
  <div class="v-input__control">
    <div class="v-field v-field--appended v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr" role="combobox">
      <div class="v-field__overlay"></div>
      <div class="v-field__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="width: 100%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
          <!---->Select Label
        </label><label class="v-label v-field-label" for="input-0">
          <!---->Select Label
        </label>
        <!---->
        <div class="v-field__input" data-no-activator="">
          <!---->
          <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" inputmode="none" aria-label="Open" title="Open">
        </div>
        <!---->
      </div>
      <!---->
      <div class="v-field__append-inner">
        <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon" aria-hidden="true"></i>
        <!---->
      </div>
      <div class="v-field__outline">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div class="v-input__details">
    <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
      <!---->
    </transition-group-stub>
    <!---->
  </div>
</div>"
`;

exports[`BaseSelect.vue > renders props.text when an empty string passed 1`] = `
"<div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field v-select v-select--single">
  <!---->
  <div class="v-input__control">
    <div class="v-field v-field--appended v-field--center-affix v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr" role="combobox">
      <div class="v-field__overlay"></div>
      <div class="v-field__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="width: 100%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class="v-field__field" data-no-activator="">
        <!----><label class="v-label v-field-label" for="input-0">
          <!---->
        </label>
        <!---->
        <div class="v-field__input" data-no-activator="">
          <!---->
          <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" inputmode="none" aria-label="Open" title="Open">
        </div>
        <!---->
      </div>
      <!---->
      <div class="v-field__append-inner">
        <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon" aria-hidden="true"></i>
        <!---->
      </div>
      <div class="v-field__outline">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div class="v-input__details">
    <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
      <!---->
    </transition-group-stub>
    <!---->
  </div>
</div>"
`;
